src/shared/components/modal/createModalLauncher.tsx

```diff
@@ -0,0 +1,35 @@
+import * as React from 'react';
+import { Modal, ModalProps } from '@patternfly/react-core';
+
+export type ModalComponentProps = Omit<ModalProps, 'children' | 'isOpen' | 'appendTo'> & {
+  ref?: React.LegacyRef<Modal>;
+};
+
+export type ModalLauncher = (onClose: () => void) => React.ReactElement;
+
+export type CreateModalLauncher = <P extends ModalComponentProps>(
+  C: React.ComponentType<P>,
+  args?: ModalComponentProps & {
+    'data-testid': string;
+  },
+) => (props: P) => ModalLauncher;
+
+export const createModalLauncher: CreateModalLauncher =
+  (Component, args) => (props) => (onClose) => {
+    const handleClose = () => {
+      props.onClose && props.onClose();
+      onClose();
+    };
+    return (
+      <Modal
+        aria-label="modal"
+        {...args}
+        {...props}
```

Comment:
Why do we need to pass down component props to PF modal? Shouldn't `args` cover all the modal props?